src/web.rs

```diff
@@ -91,6 +147,39 @@ impl WebImpl {
             ImageData::new_with_u8_clamped_array(Clamped(&bitmap), width.into()).unwrap();
 
         // This can only throw an error if `data` is detached, which is impossible.
-        self.ctx.put_image_data(&image_data, 0.0, 0.0).unwrap();
+        match self {
+            WebImpl::Canvas { ctx, .. } => ctx.put_image_data(&image_data, 0.0, 0.0).unwrap(),
+            WebImpl::OffscreenCanvas { ctx, .. } => {
+                ctx.put_image_data(&image_data, 0.0, 0.0).unwrap();
+                ctx.commit();
```

Comment:
Specifically for `OffscreenCanvas` I added calling `commit()` at the end of `set_buffer()` because otherwise nothing happens if `OffscreenCanvas` is used as a remote handle to an actual canvas, not sure if we want to leave that here or hide it behind another method.

Reply:
Actually I just tried that users can do that on their own by just creating a separate context and calling `commit()` on that.

I found myself wanting to draw some text on the canvas with the browser API so I needed **not** to commit after calling `set_buffer()`.

Reply:
So apparently `commit()` is deprecated, see https://github.com/whatwg/html/pull/3872.
It's not documented in the MDN, but apparently it works in both Firefox and Chrome, in Chrome `commit()` was actually removed entirely.